compiler/parser/src/parser/type_.rs

```diff
@@ -83,6 +83,8 @@ impl<N: Network> ParserContext<'_, N> {
     /// Also returns the span of the parsed token.
     pub fn parse_type(&mut self) -> Result<(Type, Span)> {
         if let Some(ident) = self.eat_identifier() {
```

Comment:
Does it make sense to move the call to `check_identifier` into `eat_identifier`?